### PR TITLE
python api with tests and resources

### PR DESCRIPTION
**Pull request by sdavid98**

### CODE REVIEW

File: api/tests/api_test.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+def app():
+    yield api
+
+
+@pytest.fixture()
+def client(app):
+    return app.test_client()
+
+
+def test_no_resource_given(client):
```
Comment (sgalhs) [inline api/tests/api_test.py:15]: An improved name for this test could be:
```suggestion
def test_status_code_when_no_resource_given(client):
```
or alternatively

```suggestion
def test_when_no_resource_given_then_status_code_404(client):
```
or something similar.

File: api/tests/api_test.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+
+def test_resource_not_found(client):
+    response = client.get("/non-existant-filename")
+    assert response.status_code == 404
+
+
+def test_resource_returns_whole_json(client):
+    response = client.get("/amp_organization_applications")
+    assert response.status_code == 200
```
Comment (sgalhs) [inline api/tests/api_test.py:27]: I believe this assertion deserves its own test case.

File: api/tests/api_test.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    first_element = response.json[0]
+    assert first_element == {
```
Comment (sgalhs) [inline api/tests/api_test.py:33]: Similar to the previous assertion, this may deserve its own test case. :)

File: api/tests/api_test.py
Comment (sgalhs) [inline api/tests/api_test.py:1]: Nice tests overall! (I hate the word "overall" in this context, because it makes you think, that the reviewer is not fully satisfied - when in fact I am! ;) )

Another thing that one might consider is to test what happens when `limit` and `offset` are set in such a way that the result is either empty or contains less elements than limit.

File: api/server.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+def get_path(filename):
+    return 'csv/' + filename + '.csv'
+
+
+def csv_exists(filename):
+    return path.isfile(get_path(filename))
+
+
+def read_csv(filename, limit=-1, offset=0):
+    limit = limit or -1
```
Comment (sgalhs) [inline api/server.py]: This will be somewhat of a nit-pick: -1 and 0may be extracted into variables.
Also, you're providing both with default values in the function signature, so setting default values for them shouldn't be necessary here.

File: api/tests/csv_handling_test.py
Comment (sgalhs) [inline api/tests/csv_handling_test.py:1]: I believe most of the tests could be moved into `api_test.py`. If for nothing else, then for the fact that all the functions that we're testing here are implemented in `server.py`. I could see them belonging here if the _utilities_ being tested here were actually separated into their own file, let's say. :)

File: api/tests/csv_handling_test.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        self.assertEqual('csv/some_filename.csv',
+                         server.get_path('some_filename'))
+
+    def test_csv_exists(self):
+        with self.subTest():
+            self.assertTrue(server.csv_exists('amp_applications'))
+        with self.subTest():
+            self.assertFalse(server.csv_exists('non_existent_file'))
+
+    def test_csv_parsing(self):
```
Comment (sgalhs) [inline api/tests/csv_handling_test.py]: This test case in particular is very-very similar to one that's already in `api_test.py`. I'm thinking `test_resource_returns_partial_json`. I'm only pointing this out to underline the fact that these tests could very well belong in `api_test.py`.

The fact that they are written is commendable nevertheless!

**Review by sgalhs — COMMENTED** [review]
Nice job!

File: api/server.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    offset = offset or 0
```
Comment (bnagyhearsay) [inline api/server.py]: Using `or` as a null-fallback mechanism for non-boolean values is a bad practise as it can lead to unintended consequences. E.g.:
```Python
def fun(min_items=None):
  min_items = min_items or 5
```
In this case, there is no way to pass 0 as argument as it would fall back to 5, because 0 is falsy.

A better and more explicit way to do such a fallback is to use the inline if-else, even if it is more verbose:
```Python
min_items = min_items if min_items is not None else 5
```
